### ISSUE TITLE
Client picks 0.11.0a2 over 0.11.0 when using the ` --task-name` option without a version specified

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal#549]
The current 1.3.4 client defaults to using the 0.11.0a2 pre-release tasks over the 0.11.0 released tasks when the task is specified as e.g. `fractal --batch workflow add-task $PRJ_ID $WF_ID --task-name "Convert Yokogawa to OME-Zarr"` 

I have removed the .cache folder, the behavior remains the same. It would be great to improve the heuristics of task version picking for the CLI client, so that it always picks the most recent available task without having to use the `--task-version` option

**Comment by tcompa** [comment]
This is the relevant line:
https://github.com/fractal-analytics-platform/fractal/blob/main/fractal/cmd/_aux_task_caching.py#L156C1-L156C80
together with
```python
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> "0.11.0" > "0.11.0a2"
False
```

**Comment by imagejan** [comment]
```
>>> from packaging import version
>>> version.parse("0.11.0a2") < version.parse("0.11.0")
```

Or use https://python-semver.readthedocs.io/en/latest/usage/compare-versions.html which should even work with `max()`.

**Comment by tcompa** [comment]
For the record, we should pick one reference for the definition of a version. The same exact discussion already took place in https://github.com/fractal-analytics-platform/fractal-web/issues/149#issuecomment-1614329655, since the SemVer specs (implemented e.g. in https://python-semver.readthedocs.io/en/latest/index.html or https://www.npmjs.com/package/semver) differ from the Python version specs (defined in PEP 440).

Luckily the difference does not apply to `X.Y.Z` releases, so that our only issue here is that we should not fail in parsing the version strings - and then we don't have a very strong opinion about sorting `1.0.0c2` with respect to `1.0.0alpha3`.

The choice in fractal-web was to use the [`loose` parameter of JavaScript semver package](https://github.com/npm/node-semver#functions). The python semver package doesn't include a similar parameter (only [`optional_minor_and_patch`](https://python-semver.readthedocs.io/en/latest/usage/parse-version-string.html), which is not enough) and then easily results in parsing errors (e.g. for `0.10.0c2`). The `packaging.version` one seems to accept more inputs, and then I will proceed with this one for the moment.

**Comment by tcompa** [comment]
This is in principle fixed with `1.3.5a0`.
I'll bump to `1.3.5` (and close this issue) as soon as one of us can confirm it works as expected.

**Comment by tcompa** [comment]
I confirm that the new behavior matches with the expected one.

```
# fractal-client 1.3.4

fractal workflow add-task 1 1 --task-name "Create OME-Zarr structure"
--> add task with source "pip_remote:fractal_tasks_core:0.11.0a2:fractal-tasks::create_ome-zarr_structure"

# fractal-client 1.3.5a0

fractal workflow add-task 1 1 --task-name "Create OME-Zarr structure"
--> add task with source "pip_remote:fractal_tasks_core:0.11.0:fractal-tasks::create_ome-zarr_structure"
```

**Comment by tcompa** [comment]
(now available in 1.3.5)
